src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Workspace/Mef/MefWorkspaceServices.cs

```diff
@@ -208,15 +208,10 @@ public override IEnumerable<TLanguageService> FindLanguageServices<TLanguageServ
         internal bool TryGetLanguageServices(string languageName, out MefLanguageServices languageServices)
             => _languageServicesMap.TryGetValue(languageName, out languageServices);
 
-        internal sealed class LazyServiceMetadataDebuggerProxy
+        internal sealed class LazyServiceMetadataDebuggerProxy(ImmutableArray<Lazy<IWorkspaceService, WorkspaceServiceMetadata>> services)
         {
-            private readonly ImmutableArray<Lazy<IWorkspaceService, WorkspaceServiceMetadata>> _services;
-
-            public LazyServiceMetadataDebuggerProxy(ImmutableArray<Lazy<IWorkspaceService, WorkspaceServiceMetadata>> services)
-                => _services = services;
-
             public (string type, string layer)[] Metadata
-                => _services.Select(s => (s.Metadata.ServiceType, s.Metadata.Layer)).ToArray();
+                => services.Select(s => (s.Metadata.ServiceType, s.Metadata.Layer)).ToArray();
```

Comment:
📝 I really don't like that the fields are breaking naming conventions.

Reply:
They're not fields (seriously.  the language does not think of them as fields, or treat them in any way as fields).  They are captured parameters :)

Reply:
They are being used as a replacement for a field of the same purpose, so they are functionally equivalent. I predict this is going to be problematic for feature adoption due to the amount of code that needs to change as part of this (all references to a field with a `_` prefix need to change in source control when they start referring to the captured parameter instead). Personally, I would rather not have primary constructors than to have the captures appear with this name.